osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -27,21 +28,47 @@ namespace osu.Game.Rulesets.Mods
 {
     public abstract class ModFlashlight : Mod
     {
+        private const int min_change_size_combo = 1;
+        private const int max_change_size_combo = 300;
+
         public override string Name => "Flashlight";
         public override string Acronym => "FL";
         public override IconUsage? Icon => OsuIcon.ModFlashlight;
         public override ModType Type => ModType.DifficultyIncrease;
         public override LocalisableString Description => "Restricted view area.";
 
+        protected ModFlashlight()
+        {
+            FinalFlashlightSize.DefaultChanged += _ => FinalFlashlightSize.SetDefault();
+        }
+
         [SettingSource("Flashlight size", "Multiplier applied to the default flashlight size.")]
         public abstract BindableFloat SizeMultiplier { get; }
 
-        [SettingSource("Change size based on combo", "Decrease the flashlight size as combo increases.")]
-        public abstract BindableBool ComboBasedSize { get; }
+        [SettingSource("Final flashlight size", "The final size multiplier that is reached when the flashlight changes size for the last time.")]
+        public BindableFloat FinalFlashlightSize { get; } = new BindableFloat(0.8f)
+        {
+            MinValue = 0.5f,
+            MaxValue = 1,
+            Precision = 0.1f,
+        };
+
+        [SettingSource("Change size combo", "Changes the combo multiplier where the flashlight size is changed.")]
+        public BindableInt ChangeSizeCombo { get; } = new BindableInt(100)
+        {
+            MinValue = min_change_size_combo,
+            MaxValue = max_change_size_combo,
+        };
+
+        [SettingSource("Change size times", "Changes how many times the flashlight size is changed before reaching the final flashlight size.")]
```

Comment:
Done.

Reply:
Changes is a countable noun so this should be number instead of amount, or maybe just "Size changes" and "How many times the flashlight size changes.".

Reply:
I feel like "size changes" isn't very descriptive but I agree with changing the description.

Using "number of size changes" is also fine with me

Reply:
This isn't used anymore look at recent commits.